jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -386,12 +420,21 @@ public static class ConcurrentEntry<E> implements Entry<E>
         // Other threads accessing must check the state field above first, so a good before/after
         // relationship exists to make a memory barrier.
         private E pooled;
+        // A weak ref to the pooled object; the hard ref gets nulled out when it is read
```

Comment:
The comment at line 419 is now wrong as it says the hard reference is "set once and never changed".  This is no longer true.    I would reverse the order of these fields, as the weak reference is the primary reference that is set once and never changed.  The hard reference is used to keep a released buffer in the pool.  Also since `getPooled` returns the weak reference, perhaps the names should change to:
```java
private WeakReference<E> pooled;
private E released;
```

Reply:
Agreed. And `pooled` should probably be volatile now.